zulip/integrations/matrix/matrix_bridge.py

```diff
@@ -77,30 +79,64 @@ def _matrix_to_zulip(room: Any, event: Dict[str, Any]) -> None:
 
     return _matrix_to_zulip
 
-def get_message_content_from_event(event: Dict[str, Any]) -> str:
+def get_message_content_from_event(event: Dict[str, Any]) -> Optional[str]:
+    irc_nick = shorten_irc_nick(event['sender'])
     if event['type'] == "m.room.member":
+        return None
+        # uncomment the lines below and delete the above `return`
+        # to display joining and leaving messages
+        """
         if event['membership'] == "join":
-            content = "{0} joined".format(event['sender'])
+            content = "**{0}** joined".format(irc_nick)
         elif event['membership'] == "leave":
-            content = "{0} quit".format(event['sender'])
+            content = "**{0}** quit".format(irc_nick)
+        """
     elif event['type'] == "m.room.message":
         if event['content']['msgtype'] == "m.text" or event['content']['msgtype'] == "m.emote":
-            content = "{0}: {1}".format(event['sender'], event['content']['body'])
+            content = "**{0}**: {1}".format(irc_nick, event['content']['body'])
     else:
         content = event['type']
     return content
 
+def shorten_irc_nick(nick: str) -> str:
+    """
+    Add nick shortner functions for specific IRC networks
+    Eg: For freenode change '@freenode_user:matrix.org' to 'user'
+    Check the list of IRC networks here:
+    https://github.com/matrix-org/matrix-appservice-irc/wiki/Bridged-IRC-networks
+    """
+    # for freenode
+    freenode = re.match('@freenode_([a-zA-Z0-9-_]+):matrix.org', nick)
+    if freenode:
+        return freenode.group(1)
+    # for moznet
+    moznet = re.match('@mozilla_([a-zA-Z0-9-_]+):matrix.org', nick)
+    if moznet:
+        return moznet.group(1)
+    # for hackint
+    hackint = re.match('@irc_([a-zA-Z0-9-_]+):irc.hackint.org', nick)
+    if hackint:
+        return hackint.group(1)
+    # for ircnet
+    ircnet = re.match('@_ircnet_([a-zA-Z0-9-_]+):irc.snt.utwente.nl', nick)
+    if ircnet:
+        return ircnet.group(1)
+    # for matrix users
+    matrix = re.match('@([a-zA-Z0-9-_]+):matrix.org', nick)
+    if matrix:
+        return matrix.group(1)
+    return nick
+
 def zulip_to_matrix(config: Dict[str, Any], room: Any) -> Callable[[Dict[str, Any]], None]:
-    site_without_http = config["site"].replace("https://", "").replace("http://", "")
 
     def _zulip_to_matrix(msg: Dict[str, Any]) -> None:
         """
         Zulip -> Matrix
         """
         message_valid = check_zulip_message_validity(msg, config)
         if message_valid:
-            matrix_username = zulip_to_matrix_username(msg["sender_full_name"], site_without_http)
-            matrix_text = "{0}: {1}".format(matrix_username, msg["content"])
+            matrix_username = zulip_to_matrix_username(msg["sender_full_name"])
+            matrix_text = "{0} {1}".format(matrix_username, msg["content"])
```

Comment:
Could this formatting be specified in the config too? Eg. `"message_template": "<{username}> {message}"` in the matrix section, and `"message_template": "**{username}**: {message}" for zulip. Or perhaps "content_prefix", to skip the common ` {message}` part?

Even if not in config, it could be extracted into standard variables/strings in this file, to make it clearer?